src/secontext.c

```diff
@@ -161,44 +168,110 @@ selinux_getfilecon(struct tcb *tcp, const char *path, char **secontext,
 	if (!proc_pid)
 		return -1;
 
-	int rc = -1;
+	/*
+	 * The path may contain .../proc/self/... or .../proc/self<eos>.
+	 * We need to replace 'self' by the target PID, or else we would
+	 * resolve the path with 'self' being strace itself.
+	 * In some corner cases, it's impossible to guess so keep it as is if
+	 * we don't find any occurrence of /proc/self.
+	 * Finally there may be breakage if the pathname contains /proc/self
+	 * but has nothing to do with /proc.
+	 */
+
+	int rc;
 	char fname[PATH_MAX];
+	char buf[PATH_MAX + 1];
+	size_t l = strlcpy(fname, path, sizeof(fname));
```

Comment:
strlcpy() was introduced in glibc-2.38, I don't think it's portable enough, and CI confirms that.